crates/isograph_lang_parser/src/parse_iso_literal.rs

```diff
@@ -19,11 +22,44 @@ use crate::{
     ParseResultWithLocation, ParseResultWithSpan, PeekableLexer,
 };
 
-pub fn parse_iso_fetch(
-    iso_fetch_text: &str,
+pub enum IsoLiteralExtractionResult {
+    ClientFieldDeclaration(WithSpan<ResolverDeclaration>),
+    EntrypointDeclaration(WithSpan<EntrypointTypeAndField>),
+}
+
+pub fn parse_iso_literal(
+    iso_literal_text: &str,
+    definition_file_path: ResolverDefinitionPath,
+    text_source: TextSource,
+) -> Result<WithSpan<IsoLiteralExtractionResult>, WithLocation<IsographLiteralParseError>> {
+    let mut tokens = PeekableLexer::new(iso_literal_text);
+    tokens
+        .with_span(|tokens| {
+            let discriminator = tokens
+                .parse_source_of_kind(IsographLangTokenKind::Identifier)
+                .map_err(|with_span| with_span.map(IsographLiteralParseError::from))
+                .map_err(|err| err.to_with_location(text_source))?;
+            match tokens.source(discriminator.span) {
```

Comment:
Lol whoops discriminator is already an &str here! Sorry. No need to call `tokens.source`

Reply:
Fixed with [`512fa22` (#4)](https://github.com/isographlabs/isograph/pull/4/commits/512fa229076b8a510e92dd1b1ad1e8a898b9afdf)